source/server/filter_chain_manager_impl.h

```diff
@@ -244,8 +244,16 @@ class FilterChainManagerImpl : public Network::FilterChainManager,
   using DirectSourceIPsMap = absl::flat_hash_map<std::string, SourceTypesArraySharedPtr>;
   using DirectSourceIPsTrie = Network::LcTrie::LcTrie<SourceTypesArraySharedPtr>;
   using DirectSourceIPsTriePtr = std::unique_ptr<DirectSourceIPsTrie>;
-  using ApplicationProtocolsMap =
-      absl::flat_hash_map<std::string, std::pair<DirectSourceIPsMap, DirectSourceIPsTriePtr>>;
+
+  // This would nominally be a `std::pair`, but that version crashes the Windows clang_cl compiler
+  // for unknown reasons. This variation, which is equivalent, does not crash the compiler.
+  // The `std::pair` version was confirmed to crash both clang 11 and clang 12.
+  struct DirectSourceIPsPair {
```

Comment:
wow, that must have been "fun" to find :-P

Reply:
Big thank you to @wrowe and @davinci26 for helping me figure out what was going wrong and how to fix it!